### PR TITLE
Add some extentions to model

### PR DESCRIPTION
**Pull request by prhod**
Adding `push` fn  to `Collection` and `FromStr` trait on `Time` struct.

### CODE REVIEW

File: src/collection.rs [hunk 1]
```diff
@@ -178,6 +178,31 @@ impl<T> Collection<T> {
...
+    /// let mut c = Collection::new(vec![]);
+    /// let foo_idx = c.push(Obj("foo"))?;
+    /// let bar_idx = c.push(Obj("bar"))?;
+    /// assert_eq!(&c[foo_idx], &Obj("foo"));
+    /// assert_ne!(&c[foo_idx], &Obj("bar"));
+    /// # Ok(())
+    /// # }
+    /// # fn main() { run().unwrap() }
+    /// ```
+    pub fn push(&mut self, item: T) -> Result<Idx<T>> {
```
Comment (TeXitoi) [inline src/collection.rs]: This can't fail, thus just return `Idx<T>`

File: src/objects.rs [hunk 2]
```diff
@@ -568,6 +568,28 @@ impl GetObjectType for VehicleJourney {
...
+#[derive(Debug)]
+pub enum TimeError {
+    WrongFormat,
+    WrongValue,
+}
+impl std::fmt::Display for TimeError {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match *self {
+            TimeError::WrongFormat => f.write_str("TimeError::WrongFormat"),
+            TimeError::WrongValue => f.write_str("TimeError::WrongValue"),
```
Comment (TeXitoi) [inline src/objects.rs]: Display is for humans, thus "wrong format" and "wrong value" will do the job.

File: src/objects.rs [hunk 3]
```diff
@@ -584,6 +606,23 @@ impl Time {
...
 }
+impl FromStr for Time {
+    type Err = TimeError;
+    fn from_str(time: &str) -> Result<Self, Self::Err> {
+        let mut t = time.split(':');
+        let (hours, minutes, seconds) = match (t.next(), t.next(), t.next(), t.next()) {
+            (Some(h), Some(m), Some(s), None) => (h, m, s),
+            _ => return Err(TimeError::WrongFormat),
+        };
+        let hours: u32 = hours.parse().map_err(|_err| TimeError::WrongFormat)?;
```
Comment (TeXitoi) [inline src/objects.rs]: You can `impl From<std::num::ParseFloatError> for TimeError` to automate these conversions.

Comment (prhod) [reply src/objects.rs]: thanks for the hint, I added it using From<ParseIntError>

**Pull request merged**